lib/puppet/type/cisco_interface.rb

```diff
@@ -698,6 +702,206 @@ def is_to_s(value)
     end
   end # property stp_vlan_port_priority
 
+  ###########################
+  # private vlan attributes #
+  ###########################
+
+  newproperty(:switchport_mode_private_vlan_host) do
+    desc 'Switchport private host mode of the interface.'
+
+    newvalues(
+      :host,
+      :promiscuous,
+      :disabled)
+  end # property switchport_mode_private_vlan_host
+
+  newproperty(:switchport_mode_private_vlan_host_association, array_matching: :all) do
+    format = '["primary_vlan", "secondary_vlan"]'
+    desc 'An array of [primary_vlan, secondary_vlan] pair. '\
+         "Valid values match format #{format} with primary_vlan "\
+         'and secondary_vlan as integer string.'
+    match_error = "must be of format #{format} with primary_vlan and "\
+                  'secondary_vlan as integer string.'
+
+    validate do |value|
+      fail "Vlan '#{value}' #{match_error}" unless
+            value.kind_of? String
+      fail "Vlan '#{value}' #{match_error}" unless
+            /^(\s*\d+\s*)$/.match(value).to_s == value
+    end
+
+    munge do |value|
```

Comment:
Here you can normalize the data for `primary_vlan` and `secondary_vlan` to be either strings or integers depending on what you need to match the getter.